### PR TITLE
4119, 4030

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#4119 Roxygen IntelliSense does not incrementally search
- limit trigger to `@`
- include '@' in the applicable span in comments (normally search stops on non-identifier chars)
- add tests

#4030 Update package index on connection change 
- update list of packages when connection changes

### CODE REVIEW

File: src/R/Editor/Impl/Completions/Providers/RoxygenTagCompletionProvider.cs [hunk 1]
```diff
@@ -23,13 +24,13 @@ internal sealed class RoxygenTagCompletionProvider : IRCompletionListProvider {
...
             var lineText = rawLineText.TrimStart();
+            var positionInLine = context.Position - line.Start;
+            var typedText = lineText.TextBeforePosition(positionInLine);
 
-            // Check that we are inside the Roxygen comment
-            if (!lineText.StartsWith("#'") || context.Position < rawLineText.Length - lineText.Length + 2) {
-                return completions;
+            // Check that we are inside the Roxygen comment AND just typed character is @
+            if (lineText.StartsWith("#'") && positionInLine >= 2 && typedText.StartsWithOrdinal("@")) {
+                completions.AddRange(RoxygenKeywords.Keywords.Select(k => new EditorCompletionEntry(k, k, string.Empty, _glyph)));
```
Comment (AlexanderSher) [inline src/R/Editor/Impl/Completions/Providers/RoxygenTagCompletionProvider.cs]:
```cs
RoxygenKeywords.Keywords.Select(k => new EditorCompletionEntry(k, k, string.Empty, _glyph))
```

`RoxygenKeywords.Keywords` is a predefined list. Do you want to save the whole expression as lazy field?

Comment (MikhailArkhipov) [reply src/R/Editor/Impl/Completions/Providers/RoxygenTagCompletionProvider.cs]: Sure

File: src/Windows/R/Editor/Test/Roxygen/RoxygenCompletionTest.cs [hunk 2]
```diff
@@ -33,28 +34,37 @@ public class RoxygenCompletionTest {
             filtered.Should().BeEmpty();
         }
 
-        [Test]
-        public void Keywords02() {
+        [CompositeTest]
+        [InlineData("#  f", 1)]
+        [InlineData("#'  f", 2)]
```
Comment (AlexanderSher) [inline src/Windows/R/Editor/Test/Roxygen/RoxygenCompletionTest.cs]: Do you care about construct `##`?

Comment (MikhailArkhipov) [reply src/Windows/R/Editor/Test/Roxygen/RoxygenCompletionTest.cs]: It should be same as # (i.e. no Roxygen)

**Comment by AlexanderSher** [comment]
👍

**Pull request merged**